### PR TITLE
[doc] Add missing documentation for jax.ad_checkpoint

### LINKED ISSUE #33630: jax.ad_checkpoint package does not appear in API docs
It's mentioned in the gradient checkpointing docs (https://docs.jax.dev/en/latest/gradient-checkpointing.html), so it's clearly a public API, but at least some of the exported functions (e.g., `checkpoint_name`) are not separately documented, as is standard for JAX.

### PR DESCRIPTION
**Pull request by jakevdp**
Fixes #33630. I only included `jax.ad_checkpoint.checkpoint_name` for now; `checkpoint` and `remat` are deprecated from this namespace in #33660.

#jax-fixit

### CODE REVIEW

File: jax/_src/ad_checkpoint.py [hunk 1]
```diff
@@ -373,7 +373,14 @@ def fun_remat(*args, **kwargs):
...
 
-remat = checkpoint  # alias
+
+def remat(fun: Callable, *, prevent_cse: bool = True,
+          policy: Callable[..., bool] | None = None,
+          static_argnums: int | tuple[int, ...] = (),
+          ) -> Callable:
+  """Alias of :func:`jax.ad_checkpoint.checkpoint`."""
+  return checkpoint(fun, prevent_cse=prevent_cse, policy=policy,
+                    static_argnums=static_argnums)
```
[the remark this answers is not in the document]
Comment (jakevdp) [reply jax/_src/ad_checkpoint.py]: Bad suggestion: this would rewrite the doc of `checkpoint` as well.

**Review by danielsuo — APPROVED** [review]
